run_jobs.py

```diff
@@ -61,14 +61,19 @@ def run(self) -> None:
 
         # Until we have a decorator for this
         self.started_at = time.time()
-        self.data_sources = start_method()
-        self.finished_at = time.time()
-
-        delta = self.finished_at - self.started_at
-        str_time = time.strftime('%H:%M:%S', time.gmtime(delta))
-        logger.info(f'Duration of job run: {str_time}')
-
-        self.create_metadata()
+        try:
+            self.data_sources = start_method()
+            self.finished_at = time.time()
+
+            delta = self.finished_at - self.started_at
+            str_time = time.strftime('%H:%M:%S', time.gmtime(delta))
+            logger.info(f'Duration of job run: {str_time}')
+
+            self.create_metadata()
+        except JobError as je:
+            logger.error(f'JobError: {je}')
+            logger.info(f'An error prevented the {self.name} job from finishing')
```

Comment:
I think this line can be combined with line 74 at the error level instead, since the job has been stopped due to the error.

Reply:
@zqian, I do not want these to be in the same log message. I see the first one (74) as just reporting the `JobError` message, while the second (75) is explaining how the program will respond to the error. If you want me to bump the level for the second to `ERROR`, I can do that, please tell me if that's what you want.

Reply:
I will bump this to `ERROR`, but I'd like to keep them as separate messages. I feel there's going to be too much information crammed in one line otherwise.